x/exp/batch/batch.go

```diff
@@ -375,11 +382,11 @@ func cloneSub(r types.Value, k types.String, v types.Value) (types.Value, bool)
 	return r, false
 }
 
-func findVariables(found map[types.String]struct{}, r types.Value) {
+func findVariables(found *sets.MapSet[types.String], r types.Value) {
```

Comment:
I wonder if sets.NewMapSet()... should return a pointer since the Add method leans on the ability to be able to modify things?

Reply:
Annoyingly though that costs an allocation in `in` lookups.  So I guess it's a funny tradeoff.  Hmm.